chaosazure/vmss/actions.py

```diff
@@ -48,7 +52,13 @@ def delete_vmss(filter: str = None,
             client.virtual_machine_scale_set_vms.delete(
                 scale_set['resourceGroup'],
                 scale_set['name'],
-                instance['instanceId'])
+                instance['instance_id'])
+            instances_records.add(cleanse.vmss_instance(instance))
+
+        scale_set['compute'] = instances_records.output()
```

Comment:
I'm not sure about its value.

The type is _always_ defined in the resource's dictionary object itself. When you take a look at the sample JSON and the **type** attributes of the different resources.

What would be the value of having it the way you proposed? When for example I think of parsing such an output. Is it good to have different keys that are named after the type of the resource or is it appropriate to have a "standard" key to parse them?

I am not sure as well - just thinking loudly.

Reply:
In current implementation we want to name list of nested resources: **"compute"**. My proposal was to name it **"virtualMachines"**. My thinking is following:
 - **compute** - it is a name that was defined by you. It is correct and valid - although maybe it should be plurar as it is a list.
- **virtualMachines** - this name follows the convention of Azure REST API.  Please take a look at other rest API that has nested resources e.g.: [AvailablitySet](https://docs.microsoft.com/en-us/rest/api/compute/availabilitysets/get#availabilityset). It returns **"virutalMachines:[]"** for subresources  not **"compute: []"**. 

Using "types" as a "key" for nested resources in this case "virtualMachines" should prevent in the future such conversations as this "How to name this subresources", and it seems we will be more aligned with Azure REST API way of naming it.

And another topic is how to name our top level resources. "resources" or as well use a "type". I think both has some advantages but having "resource" like now is ok for me  :)

Reply:
I was talking about the top level resource :)

Concerning the **top level resource**. I think it's absolutely okay to name it resources - because that's what it is. Later there may be additional elements such operation results, command results, and so on. In the end it will be clear for the user what he has to expect there.

Concerning the nested resource.

1) We may not forget that the resource we retrieve are coming from the resource graph and not from the Microsoft REST API.
2) The resource graph may collect it from the MS REST API but may display it different than the MS REST API docs are describing. Such an example: The Scale Set does not define any *instances* or *virtualMachines* in its [definition](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachinescalesets/get#virtualmachinescaleset). But retrieving it via the resource graph delivers you *instances*.

How do you want to name it? _Instances_ because the resource graph delivers it that way? Or virtualMachines because it's the official name in the MS REST API? From where do you take the name value - a constant from the ctk-azure code or from the resource graph response?

Reply:
Fair point with the "resource graph". What is your opinion? I guess that we can agree that we have a better options than "compute". I would consider

- instances
- virutalMachines
- resources or subResources or nestedResources (this could be use everywere no matter if we ask for compute or other types of resource like network...)

Reply:
Put it virtualMachines